compiler/rustc_monomorphize/src/partitioning.rs

```diff
@@ -311,7 +311,7 @@ fn merge_codegen_units<'tcx>(
     assert!(cx.tcx.sess.codegen_units().as_usize() >= 1);
 
     // A sorted order here ensures merging is deterministic.
-    assert!(codegen_units.is_sorted_by(|a, b| Some(a.name().as_str().cmp(b.name().as_str()))));
+    assert!(codegen_units.is_sorted_by(|a, b| a.name().as_str() < b.name().as_str()));
```

Comment:
Shouldn't this be <=?

As-is this is a change in behavior - maybe a correct one but I'd prefer behavior changes in a separate PR (or at least commit).

(Same applies in one more case above).

Reply:
Good catch, fixed in c2c00f5